### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by kbonney**
Changelog:

- Updating fatigue utilities.
- Added docstring to shellMeshGeneral
- updating gitignore for ansys outputs
- Cleaning up object files (getting rid of old comments, formatting, docstrings)
- Cleaning/refactoring of ANSYS files
- Changing `blade.components` and `blade.materials` to dictionaries indexed by names, instead of lists.
- Cleaned up example folder by removing unnecessary files and reorganizing.
- Added tests for Ansys and Mesh functionality.
- Added plotting for airfoils `af.plotAirfoil()` and components `component.plotcp()`
- Adjusted creation of matdb to account for components and materials changing from list structure to dict structure.
- Moved shell generation methods out of blade object. They are now standalone functions in shell.py
- Changed blade.mesh attribute to be a parameter passed into mesh functions now called elementSize.

### CODE REVIEW

**Comment by kbonney** [comment]
I am seeing errors when running the meshing examples. Going to hold off on pulling until resolved.

File: pynumad/io/yaml_to_blade.py [hunk 1]
```diff
@@ -96,8 +94,8 @@ def yaml_to_blade(blade, filename: str, write_airfoils: bool = False):
     
     # Leading and Trailing Edge bands are constants in millimeters
 
-    blade.leband = np.array(blade_internal_structure['layers'][I_LE]['width']['values'])*1000 / 2
```
Comment (ecamare) [inline pynumad/io/yaml_to_blade.py:89]: Kirk, we need to keep these as arrays instead of constants.

File: pynumad/objects/Blade.py [hunk 2]
```diff
@@ -548,25 +503,21 @@ def updateKeypoints(self):
...
             # width and offset [HP LP]
-            # if len(self.sparcapwidth) > 2 or len(self.sparcapoffset) > 2:
-            #     raise Exception('too many entries for spar cap definition')
-            # scwidth_hp = mm_to_m * self.sparcapwidth[0] #type: float
-            # scwidth_lp = mm_to_m * self.sparcapwidth[-1] #type: float
-            n1 = mm_to_m * self.leband[k] # no foam width
-            n2 = mm_to_m * self.teband[k] # no foam width
-
-
-            scwidth_hp = mm_to_m * self.sparcapwidth_hp[k] #type: float
```
Comment (ecamare) [inline pynumad/objects/Blade.py]: The spar cap widths should be variable along the span; not constant. We need to revert back to: 

scwidth_hp = mm_to_m * self.sparcapwidth_hp[k]

File: pynumad/io/yaml_to_blade.py (hunk 1 above)
Comment (kbonney) [reply pynumad/io/yaml_to_blade.py:89]: Okay I wasn't sure if that was apart of the changes Evan made. I'll pull it back in.

**Pull request merged**